extensions/llama/typescript/llama.ts

```diff
@@ -0,0 +1,343 @@
+import {
```

Comment:
The functions are missing callback events.

Would recommend following the pattern in openai.ts:
```
    const startEvent = {
      name: "on_serialize_start",
      file: __filename,
      data: {
        promptName,
        data,
        params,
      },
    } as CallbackEvent;
    aiConfig.callbackManager.runCallbacks(startEvent);
```

This more broadly shows a gap where every model parser creator will have to do the right think with callback events. Perhaps we modify the interface where we have abstract implementations for `run`, `deserialize` and `serialize` which wrap a `runImpl` etc around the callback events, so it's consistent across all model parsers.

cc @LastMileNewGrad @jonathanlastmileai (same for python side)

Reply:
@saqadri this is why I mentioned exporting them -- they aren't exposed in the package for me to actually add them :)

Reply:
I think we can do that for these types of callbacks since they all seem to be before/after events around the method implementations anyway